src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -36,12 +128,30 @@
     }
 
     @Override
-    public int size() {
-        return 0;
+    public T get(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException(
+                    "Index outside the list");
+        }
+
+        Node<T> current;
+        if (index < size / 2) {
+            current = head;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+        } else {
+            current = tail;
+            for (int i = size - 1; i > index; i--) {
+                current = current.prev;
+            }
+        }
+
+        return current.data;
     }
 
     @Override
-    public boolean isEmpty() {
-        return false;
+    public T set(T value, int index) {
+        return null;
```

Comment:
The `set` method is not implemented. It should replace the element at the specified position in this list with the specified element.